### ISSUE TITLE
Add isIsoDate predicate

### ISSUE THREAD

**Issue by jcamiel** [Orange-OpenSource/hurl#2427]
The current [`isDate`](https://hurl.dev/docs/asserting-response.html#predicates) predicate takes "date" type as input. To use it on a string, user has to explicitly convert it to a date using a format spec and [`toDate` filter](https://hurl.dev/docs/filters.html#todate):

```hurl
POST https://foo.com
HTTP 200
[Asserts]
jsonpath "$.created" toDate "%Y-%m-%dT%H:%M:%S%.f%:z" isDate
```

As the user must explicitly transform the string to a date, `isDate` has little value.

## Proposal

Add a `isIsoDate` predicate that take a string as input and check that this string can be converted to a date with the format  spec `%Y-%m-%dT%H:%M:%S%.f%:z`(chrono format). 

> Like [`Date.parse`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse) and [`Date.propotype.toISOString`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toString) JavaScript methods, we're only checking the [JavaScript date time string format](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_time_string_format) `YYYY-MM-DDTHH:mm:ss.sssZ` (simplicification of the ISO 8601)

```hurl
POST https://foo.com
HTTP 200
[Asserts]
jsonpath "$.created" isIsoDate
```

**Comment by beaterblank** [comment]
I'd like to take this issue, very similar to `isDate` predicate so I should be able to implement a similar flow.

**Comment by beaterblank** [comment]
`NaiveDateTime::parse_from_str`  with `%Y-%m-%dT%H:%M:%S%.f%:z"` does not work for some reason but removing the `%` after `f` makes it work.

Also what about the time format `%Y-%m-%dT%H:%M:%S%.fZ` ? should we also consider this as an iso date while testing?

I was thinking of using this high level logic : 

```{rust}
use chrono::NaiveDateTime;

fn main() {
    const DATE_FMTS : [&str;2] = ["%Y-%m-%dT%H:%M:%S%.f:z","%Y-%m-%dT%H:%M:%S%.fZ"];
    for fmt in DATE_FMTS{
        if let Ok(_) = NaiveDateTime::parse_from_str(" 2024-02-15T19:47:46.266:z",fmt){
            println!("matched with {}",fmt);
        }
    }
}
```

**Comment by beaterblank** [comment]
Please check the PR for above things and let me know if any feedback. Thank you!

**Comment by jcamiel** [comment]
Hi @beaterblank I'll make feedback asap

**Comment by jcamiel** [comment]
Hi @beaterblank this has been fixed by #2565. I'm. closing this one now
